src/com/booksaw/betterTeams/commands/teama/TeleportTeama.java

```diff
@@ -0,0 +1,113 @@
+package com.booksaw.betterTeams.commands.teama;
+
+import com.booksaw.betterTeams.CommandResponse;
+import com.booksaw.betterTeams.Team;
+import com.booksaw.betterTeams.commands.presets.TeamSelectSubCommand;
+import com.booksaw.betterTeams.message.HelpMessage;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import java.util.List;
+
+public class TeleportTeama extends TeamSelectSubCommand {
+
+    @Override
+    public CommandResponse onCommand(CommandSender sender, String label, String[] args, Team team) {
+        Team specifiedTeam = Team.getTeam(args[0]);
+
+        if (specifiedTeam == null) {
+            return new CommandResponse("noTeam");
+        }
+
+        if (!(sender instanceof Player)) {
+            return new CommandResponse("needPlayer");
+        }
+
+        Player player = (Player) sender;
+        Location location = player.getLocation(); // using the sender's location as a default
```

Comment:
Instead of having the repeated teleport code for each case (/teama teleport, /teama teleport <x> <y> <z>

Use the location variable to store where you are teleporting the players to then have the teleport code at the end of the args checks.